### ISSUE TITLE
SGD Training results in NaN-Outputs

### ISSUE THREAD

**Issue by FelixVoelker** [deeplearning4j/deeplearning4j#1810]
Simple x^2 regression task, trained with Stochastic Gradient Descent without minibatch, returns only NaN values. Seems like the updater always applies the minibatch size on the gradient, how can I set it on 1? 

` MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
                .seed(System.currentTimeMillis())
                .iterations(5000)
                .miniBatch(false)
                .learningRate(0.01)
                .weightInit(WeightInit.XAVIER)
                .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
                .updater(Updater.SGD)
                .list()
                .layer(0, new DenseLayer.Builder().nIn(1).nOut(500).activation("tanh").build())
                .layer(1, new DenseLayer.Builder().nIn(500).nOut(250).activation("tanh").build())
                .layer(2, new DenseLayer.Builder().nIn(250).nOut(100).activation("tanh").build())
                .layer(3, new OutputLayer.Builder().nIn(100).nOut(1).activation("identity").lossFunction(LossFunctions.LossFunction.MSE).build())
                .pretrain(false)
                .backprop(true)
                .build();

MultiLayerNetwork nn = new MultiLayerNetwork(conf);
nn.init();

int nSamples = 21;
INDArray x0 = Nd4j.linspace(-10, 10, nSamples).reshape(nSamples,1);
INDArray x1 = x0.mul(x0);

nn.fit(x0, x1);
System.out.println(nn.output(x0, false));`
